rllib/core/learner/torch/torch_learner.py

```diff
@@ -224,4 +231,5 @@ def _map_module_to_device(self, module: MultiAgentRLModule) -> None:
             module.to(self._device)
         else:
             for key in module.keys():
-                module[key].to(self._device)
+                if isinstance(module[key], torch.nn.Module):
+                    module[key].to(self._device)
```

Comment:
Because of this, could a user theoretically write an RLModule themselves and forget to inherit from torch.nn and have full training working and everything but always have paramters on the "wrong" device?

Reply:
+1 to artur's first point, but idk about the second one. why would the params be on the wrong device?

Reply:
Just asking because I don't have the full picture here.
If we only move params to the device of the learner if the RLModule inherits from nn.Module, what would happen if I were to write an RLModule that does not inherit from nn.Module but do the rest correctly?

Reply:
Yes, that's possible. But we will error out later during update, if we don't find any trainable module that doesn't have parameters (if parameter list ends up being empty). The only way we can call `to` on a module or get its parameters is when the module inherits from nn.Module or in tf case from tf.keras.Model. To give you a sense of where this check is useful consider a MARL case where once of the modules is random. You don't want to apply gradient based updates to your random RLModule, and it's also not a nn.Module or keras.Module anyway.

Reply:
If you have other kind of module that don't fit this, you can always extend your own Learner and pass it in. This is the constraint on TorchLearner and TfLearner. So it's not a problem.